### ISSUE TITLE
[Bug]: rctcbc_gov_uk returning extra incorrect dates 

### ISSUE THREAD

**Issue by undulattice** [mampfes/hacs_waste_collection_schedule#1916]
### I Have A Problem With:

A specific source

### What's Your Problem

I've been using this integration for a while without issue.
I've been away for a few weeks, and come home to find 3 entries for each entry for each week, when there should only be one.

See attached screenshot.
The Wednesday entries are the only valid ones. Not sure why they're being duplicated though. :-/

<img width="1184" alt="Screenshot 2024-03-28 at 03 44 55" src="https://github.com/mampfes/hacs_waste_collection_schedule/assets/63560223/7cecf6a6-9ee5-42cb-875f-50d91877ceeb">

### Source (if relevant)

rctcbc_gov_uk

### Logs

```Shell
no relevant logs
```

### Relevant Configuration

```YAML
waste_collection_schedule:
  sources:
    - name: rctcbc_gov_uk
      args:
        uprn: "200003766278"
      calendar_title: "Bin Collection"
```

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [X] Checked that the website of your service provider is still working
- [X] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [X] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by neildotwilliams** [comment]
I am having the same issue. Going to the council website and the calendar looks ok:
![image](https://github.com/mampfes/hacs_waste_collection_schedule/assets/8913897/d58527c4-9e1b-4db4-a06e-e9bcee31591b)

https://www.rctcbc.gov.uk/EN/Resident/RecyclingandWaste/RecyclingandWasteCollectionDays.aspx?uprn=<xxxx>&month=0

**Comment by 5ila5** [comment]
Fixed by #1917 

This will be part of release 1.47.0 or is available now as master (git) version (3-dot menu -> redownload -> change version to master -> fully restart HA (this will increase update cycle by a lot)

**Comment by neildotwilliams** [comment]
@5ila5 I have just updated this component by following your instructions and it is working well for me. 

Thanks very much, I appreciate you spending your time on this and fixing this it so quickly.
